torch/distributed/distributed_c10d.py

```diff
@@ -487,7 +496,15 @@ def _new_process_group_helper(world_size,
             _pg_map[pg] = (Backend.NCCL, store)
             _pg_names[pg] = group_name
         else:
-            raise RuntimeError("Unsupported distributed backend by group")
+            module_name, instantiating_interface_name = getattr(Backend, backend.upper())
+            other_backend = sys.modules[module_name]
```

Comment:
This lookup and the `getattr` below shouldn't be necessary.

Reply:
The problem is here we couldn't remove getattr. I will explain on the formal reply.